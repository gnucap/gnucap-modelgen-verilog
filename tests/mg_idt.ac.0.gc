attach ./modelgen_0.so
attach ../vams/vsine.so

verilog

`modelgen
module test_idtac0(p, q, r, s, n);
	(* desc="" *) real d;
	(* desc="" *) real i;
	electrical p, n;
	analog begin
		d = V(p,n);
		i = idt(d,0);
		I(q,n) <+ i;

		I(r,n) <+ idt(d,0) + V(r,n);
		I(s,n) <+ laplace_nd(V(p, n), '{1.}, '{0., 1.}) + V(s,n);
	end
endmodule

!make test_idtac0.so > /dev/null
attach ./test_idtac0.so

verilog

test_idtac0 #() d0(1, 2, 3, 4, 0);
resistor #(1.) r (2,0);
vsine #(.mag(1.)) v1(1, 0);

dc // BUG?
print ac vm(nodes) vp(nodes)
ac {1./3.1415/2.}
ac 1
status notime
end
