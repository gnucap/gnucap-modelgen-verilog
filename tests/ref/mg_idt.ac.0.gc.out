#          
 0.        
#Freq       vm(1)      vm(2)      vm(3)      vm(4)      vp(1)      vp(2)      vp(3)      vp(4)     
 0.15916    1.         0.99997    0.99997    0.99997    0.         90.        90.        90.       
#Freq       vm(1)      vm(2)      vm(3)      vm(4)      vp(1)      vp(2)      vp(3)      vp(4)     
 1.         1.         0.15915    0.15915    0.15915    0.         90.        90.        90.       
Gnucap   System status
iterations: op=0, dc=3, tran=0, fourier=0, total=5
transient timesteps: accepted=0, rejected=0, total=0
nodes: user=4, subckt=0, model=4, total=8
dctran density=53.1%, ac density=43.8%
